earth_enterprise/src/fusion/autoingest/AssetFactory_unittest.cpp

```diff
@@ -51,25 +61,106 @@ class MockAssetStorage
               }
 };
 
+
+// added ID field to aid in comparison
 class MockAssetConfig
 {
 public:
     int8_t ID;
-    MockAssetConfig(int8 _ID = -1) : ID(_ID){}
+    MockAssetConfig(int8 _ID = -1) : ID(_ID) {}
     bool operator==(const MockAssetConfig& other) const { return ID == other.ID; }
     bool operator==(uint8_t _ID) { return ID == _ID; }
     void operator=(const MockAssetConfig& other) { ID = other.ID; }
     void operator=(uint8_t _ID) { ID = _ID; }
  };
 
+class MockAssetVersionImpl : public MockAssetStorage
+{
+public:
+    using MutableAssetType = MockMutableAsset;
+    static AssetDefs::Type EXPECTED_TYPE;
+    static string EXPECTED_SUBTYPE;
+
+    using Base = MockAssetStorage;
+
+    MockAssetConfig config;
+    bool needed;
+
+    void Modify(const khMetaData& _meta, const MockAssetConfig& _config)
+    {
+        meta = _meta;
+        config = _config;
+    }
+
+    void Modify(const vector<SharedString>& _inputs,
+                const khMetaData& _meta,
+                const MockAssetConfig& _config)
+    {
+        inputs = _inputs;
+        Modify(_meta, _config);
+    }
+
+    MockAssetVersionImpl(MockAssetStorage storage,
+                  const MockAssetConfig &config_)
+                : MockAssetStorage(storage), config(config_), needed(false) {}
+};
+
+
+class MockMutableAssetVersion : public MockAssetStorage
```

Comment:
I don't think `MockMutableAssetVersion` should be deriving from `MockAssetStorage` since it's mocking a handle. I think it should just not derive from anything (like `MockMutableAsset` is doing below).